scripts/ui/display.py

```diff
@@ -99,20 +98,27 @@ def __register_model_callbacks(self, id):
             partial(self.mav_event_signal.emit, event_type, id))
 
     def __start_transport(self):
+
         if self._transport is not None and self._transport.isOpen():
             self._transport.close()
         self.mav_event_signal.connect(self.__mav_event_handler)
         if self.config.connection_mode == config.ConnectionMode.TOWER:
             self._transport = RCTTCPServer(self.port_val, self.connection_handler)
             self._transport.open()
         else:
-            try:
-                self._transport = RCTTCPClient(addr=self.addr_val, port=self.port_val)
-                self.connection_handler(self._transport, 0)
-            except ConnectionRefusedError:
-                self.user_popups.show_warning("Failure to connect:\nPlease ensure server is running.")
-                self._transport.close()
-                return
+            attempts = 5
+            retry_time = 5
+            for i in range(attempts):
+                try:
+                    self._transport = RCTTCPClient(addr=self.addr_val, port=self.port_val)
+                    self.connection_handler(self._transport, 0)
```

Comment:
Make sure only one client is created if successful--one problem I had while exploring the cause of the GCS crashing was that a second client + mav model were created but did not receive heartbeats.